### PR TITLE
fix: race condition in downing provider spec

### PR DESCRIPTION
**Pull request by johanandren**
References #31194

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Both seem valid behaviors, so :+1:

File: akka-cluster/src/test/scala/akka/cluster/DowningProviderSpec.scala [hunk 1]
```diff
@@ -76,17 +76,28 @@ class DowningProviderSpec extends AnyWordSpec with Matchers {
...
-      cluster.join(cluster.selfAddress)
+      maybeSystem.foreach { system =>
+        val cluster = Cluster(system)
+        cluster.join(cluster.selfAddress)
 
-      awaitCond(cluster.isTerminated, 3.seconds)
-      shutdownActorSystem(system)
+        awaitCond(cluster.isTerminated, 3.seconds)
+        shutdownActorSystem(system)
+      }
```
Comment (octonato) [inline akka-cluster/src/test/scala/akka/cluster/DowningProviderSpec.scala:100]: A `Try().foreach` would have been simpler

**Review by octonato — APPROVED** [review]
LGTM

File: akka-cluster/src/test/scala/akka/cluster/DowningProviderSpec.scala [hunk 2]
```diff
@@ -76,17 +76,28 @@ class DowningProviderSpec extends AnyWordSpec with Matchers {
     }
 
     "stop the cluster if the downing provider throws exception in props method" in {
-      val system = ActorSystem(
-        "auto-downing",
-        ConfigFactory.parseString("""
+      // race condition where the downing provider failure can happen fast enough
+      // that creating the actor system throws on construcing thread,
```
Comment (patriknw) [inline akka-cluster/src/test/scala/akka/cluster/DowningProviderSpec.scala:80]: The exception will cause `postStop` in `ClusterDaemon` to run coordinated shutdown, which may run ActorSystem.shutdown before other things have been initialized.

An alternative solution for this test would be to configure:
```
akka.cluster.run-coordinated-shutdown-when-down = off
```

File: akka-cluster/src/test/scala/akka/cluster/DowningProviderSpec.scala [hunk 3]
```diff
@@ -76,17 +76,28 @@ class DowningProviderSpec extends AnyWordSpec with Matchers {
...
+      val maybeSystem = try {
+        Some(
+          ActorSystem(
+            "auto-downing",
+            ConfigFactory.parseString("""
           akka.cluster.downing-provider-class="akka.cluster.FailingDowningProvider"
-        """).withFallback(baseConf))
+        """).withFallback(baseConf)))
+      } catch {
+        case _: Throwable =>
```
Comment (patriknw) [inline akka-cluster/src/test/scala/akka/cluster/DowningProviderSpec.scala:90]: `NonFatal(_)`

**Review by patriknw — APPROVED** [review]
LGTM, with alternative suggestion (don't have a strong opinion about what is best)

**Pull request merged**